synapse/handlers/e2e_keys.py

```diff
@@ -497,6 +500,19 @@ async def query_local_devices(
             local_query, include_displaynames
         )
 
+        # Check if the application services have any additional results.
+        if self._query_appservices_for_keys:
+            # Query the appservices for any keys.
+            appservice_results = await self.appservice_handler.query_keys(query)
+
+            # Merge results, overriding with what the appservice returned.
+            for user_id, devices in appservice_results.get("device_keys", {}).items():
```

Comment:
`appservice_results` is an `Iterable`, and `.get` doesn't like that

```
element-dev-stack-eledev-synapse-1      | 2023-03-24 22:15:11,675 - synapse.http.server - 123 - ERROR - POST-35 - Failed handle request via 'KeyQueryServlet': <XForwardedForRequest at 0x7f35bdd9f210 method='POST' uri='/_matrix/client/r0/keys/query' clientproto='HTTP/1.0' site='8008'>
element-dev-stack-eledev-synapse-1      | Traceback (most recent call last):
element-dev-stack-eledev-synapse-1      |   File "/usr/local/lib/python3.11/site-packages/synapse/http/server.py", line 306, in _async_render_wrapper
element-dev-stack-eledev-synapse-1      |     callback_return = await self._async_render(request)
element-dev-stack-eledev-synapse-1      |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
element-dev-stack-eledev-synapse-1      |   File "/usr/local/lib/python3.11/site-packages/synapse/http/server.py", line 516, in _async_render
element-dev-stack-eledev-synapse-1      |     callback_return = await raw_callback_return
element-dev-stack-eledev-synapse-1      |                       ^^^^^^^^^^^^^^^^^^^^^^^^^
element-dev-stack-eledev-synapse-1      |   File "/usr/local/lib/python3.11/site-packages/synapse/rest/client/keys.py", line 213, in on_POST
element-dev-stack-eledev-synapse-1      |     result = await self.e2e_keys_handler.query_devices(
element-dev-stack-eledev-synapse-1      |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
element-dev-stack-eledev-synapse-1      |   File "/usr/local/lib/python3.11/site-packages/synapse/logging/opentracing.py", line 914, in _wrapper
element-dev-stack-eledev-synapse-1      |     return await func(*args, **kwargs)  # type: ignore[misc]
element-dev-stack-eledev-synapse-1      |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
element-dev-stack-eledev-synapse-1      |   File "/usr/local/lib/python3.11/site-packages/synapse/handlers/e2e_keys.py", line 155, in query_devices
element-dev-stack-eledev-synapse-1      |     local_result = await self.query_local_devices(local_query)
element-dev-stack-eledev-synapse-1      |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
element-dev-stack-eledev-synapse-1      |   File "/usr/local/lib/python3.11/site-packages/synapse/logging/opentracing.py", line 914, in _wrapper
element-dev-stack-eledev-synapse-1      |     return await func(*args, **kwargs)  # type: ignore[misc]
element-dev-stack-eledev-synapse-1      |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
element-dev-stack-eledev-synapse-1      |   File "/usr/local/lib/python3.11/site-packages/synapse/handlers/e2e_keys.py", line 509, in query_local_devices
element-dev-stack-eledev-synapse-1      |     for user_id, devices in appservice_results.get("device_keys", {}).items():
element-dev-stack-eledev-synapse-1      |                             ^^^^^^^^^^^^^^^^^^^^^^
element-dev-stack-eledev-synapse-1      | AttributeError: 'list' object has no attribute 'get'
```

Reply:
looks like it's caused by "Patch together the results" generating a list - not sure if intended